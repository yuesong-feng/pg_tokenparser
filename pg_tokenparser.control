# pg_testgen extension
comment = 'PostgreSQL self-defined token parser'
default_version = '1.0'
module_pathname = '$libdir/pg_tokenparser'
relocatable = true
