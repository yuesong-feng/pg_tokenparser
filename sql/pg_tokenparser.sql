CREATE EXTENSION pg_tokenparser;

SELECT alias, description, token FROM ts_debug('my_cfg', 'token_01 tagset-02 tagset03 token:04');

SELECT to_tsvector('my_cfg', 'token_01 tagset-02 tagset03 token:04');

DROP EXTENSION pg_tokenparser;