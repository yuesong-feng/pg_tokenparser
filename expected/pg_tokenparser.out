CREATE EXTENSION pg_tokenparser;
SELECT alias, description, token FROM ts_debug('my_cfg', 'token_01 tagset-02 tagset03 token:04');
 alias |  description  |   token   
-------+---------------+-----------
 word  | Word          | token_01
 blank | Space symbols |  
 word  | Word          | tagset-02
 blank | Space symbols |  
 word  | Word          | tagset03
 blank | Space symbols |  
 word  | Word          | token:04
(7 rows)

SELECT to_tsvector('my_cfg', 'token_01 tagset-02 tagset03 token:04');
                     to_tsvector                      
------------------------------------------------------
 'tagset-02':2 'tagset03':3 'token:04':4 'token_01':1
(1 row)

DROP EXTENSION pg_tokenparser;
